test/test-image.sh

```diff
@@ -2,28 +2,31 @@
 
 set -e
 
+: ${CONTAINER_CLI:?Need to provide environment variable CONTAINER_CLI}
 : ${IMAGE:?Need to provide environment variable IMAGE}
 : ${EXPECTED_ARCH:?Need to provide variable EXPECTED_ARCH}
 : ${EXPECTED_VERSION:?Need to provide environment variable EXPECTED_VERSION}
 
-IMAGE_ARCH=$(docker inspect ${IMAGE} | jq -r ".[].Architecture")
+echo "CONTAINER CLI ${CONTAINER_CLI}"
+
+IMAGE_ARCH=$(${CONTAINER_CLI} inspect ${IMAGE} | jq -r ".[].Architecture")
 echo "Image architecture ${IMAGE_ARCH}"
 
 if [[ "${IMAGE_ARCH}" != "${EXPECTED_ARCH}" ]] ; then
   echo "Unexpected architecture, got ${IMAGE_ARCH}, expected ${EXPECTED_ARCH}"
   exit 1
 fi
 
-if [[ "${IMAGE_ARCH}" == $(dpkg --print-architecture) ]] ; then
-  CONTAINER_VERSION=$(docker run --rm ${IMAGE} --version)
+if [[ "${IMAGE_ARCH}" == "${EXPECTED_ARCH}" ]] ; then
```

Comment:
`dpkg` is debian specific hence this script fails on other arch hence the change

Reply:
I updated this as I was not able to test this locally on a mac machine. 
But it looks like might need a better way to handle this than this solution as test takes longer